frontend/src/app/modules/interventions/list/table/interventions.datasource.ts

```diff
@@ -0,0 +1,36 @@
+import { CollectionViewer, DataSource } from '@angular/cdk/collections';
+import { Observable, BehaviorSubject, of } from 'rxjs';
+
+import { InterventionsService } from '../../interventions.service';
+import { catchError, finalize } from 'rxjs/operators';
+
+import { Intervention, InterventionsFilter } from '@shared/domain/intervention.model';
+
+export class InterventionsDatasource implements DataSource<Intervention> {
+  private interventionSubject = new BehaviorSubject<Intervention[]>([]);
+  private loadingSubject = new BehaviorSubject<boolean>(false);
+
+  public loading$ = this.loadingSubject.asObservable();
+
+  constructor(private interventionsService: InterventionsService) {}
+
+  connect(collectionViewer: CollectionViewer): Observable<Intervention[]> {
+    return this.interventionSubject.asObservable();
+  }
+
+  disconnect(collectionViewer: CollectionViewer): void {
+    this.interventionSubject.complete();
+    this.loadingSubject.complete();
+  }
+
+  loadInterventions(params: InterventionsFilter) {
```

Comment:
It will definitely work, however you wrote it half-classical, half-reactive - I'd go with fully reactive approach :)

To make it fully reactive, the method should return the Observable, and should be Subscribed
'from the outside' when needed.

Also, all the actions would be enclosed in the pipe:

Let me write it down for you:
```
// `of` will start the pipe on Subscribe with a given initial value. 
// but it is not relevant for us, since we have the params already in scope

return of({}).pipe(
    tap(() => this.loadingSubject.next(true)), // tap is for side effects like loadingSubject, but also can be used to get the value from the stream without transforming it
    switchMapTo(this.interventionsService
               .getInterventions(params)
               .pipe(
                   tap(interventions => this.interventionSubject.next(interventions),
                // not sure if we need to catch the error - we might throw it to the console, Sentry will catch and log it on production, although I cannot really make my mind here :)
                 finalize(() => this.loadingSubject.next(false))
                 )

)

Reply:
Cool, thanks, let's do it fully reactive 😎
This needed one correction - `finalize` needed to be out of `tap`, just as part of `pipe`. Probably a typo on your side.